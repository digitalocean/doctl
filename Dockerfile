FROM alpine:3.8

ENV DOCTL_VERSION=1.17.0

RUN apk add --no-cache curl

RUN mkdir /lib64 && ln -s /lib/libc.musl-x86_64.so.1 /lib64/ld-linux-x86-64.so.2

WORKDIR /app

RUN curl -L https://github.com/digitalocean/doctl/releases/download/v${DOCTL_VERSION}/doctl-${DOCTL_VERSION}-linux-amd64.tar.gz  | tar xz

ENTRYPOINT ["./doctl"]
