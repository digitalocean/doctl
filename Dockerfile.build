FROM golang:1.14-alpine

ENV MAKE_TARGET linux_amd64

RUN  apk add --update \
     bash \
     curl \
     git \
     make \
     g++ \
     coreutils \
     libc6-compat
