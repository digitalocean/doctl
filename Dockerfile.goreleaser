FROM alpine:3.8

RUN mkdir /lib64 && ln -s /lib/libc.musl-x86_64.so.1 /lib64/ld-linux-x86-64.so.2

WORKDIR /app

COPY doctl .

ENTRYPOINT ["./doctl"]
