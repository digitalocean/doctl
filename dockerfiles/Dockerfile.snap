# Note to maintainers: after you make changes to this file, please run `make snap_image`.
FROM ubuntu:xenial as builder

RUN apt update && \
        apt dist-upgrade --yes && \
        apt install --yes curl jq squashfs-tools

# Grab the core snap from the stable channel and unpack it in the proper place
RUN curl -L $(curl -H 'X-Ubuntu-Series: 16' 'https://api.snapcraft.io/api/v1/snaps/details/core' | jq '.download_url' -r) --output core.snap && \
        mkdir -p /snap/core && \
        unsquashfs -d /snap/core/current core.snap

# Grab the snapcraft snap from the stable channel and unpack it in the proper place
RUN curl -L $(curl -H 'X-Ubuntu-Series: 16' 'https://api.snapcraft.io/api/v1/snaps/details/snapcraft?channel=stable' | jq '.download_url' -r) --output snapcraft.snap && \
        mkdir -p /snap/snapcraft && \
        unsquashfs -d /snap/snapcraft/current snapcraft.snap

# Create a snapcraft runner (TODO: move version detection to the core of snapcraft)
RUN mkdir -p /snap/bin && \
        echo "#!/bin/sh" > /snap/bin/snapcraft && \
        snap_version="$(awk '/^version:/{print $2}' /snap/snapcraft/current/meta/snap.yaml)" && \
        echo "export SNAP_VERSION=\"$snap_version\"" >> /snap/bin/snapcraft && \
        echo 'exec "$SNAP/usr/bin/python3" "$SNAP/bin/snapcraft" "$@"' >> /snap/bin/snapcraft && \
        chmod +x /snap/bin/snapcraft

# Grab the golang snap from the stable channel, unpack it in the proper place, and create a runner for it
RUN curl -L $(curl -H 'X-Ubuntu-Series: 16' 'https://api.snapcraft.io/api/v1/snaps/details/go?channel=1.13/stable' | jq '.download_url' -r) --output go.snap && \
        mkdir -p /snap/go && \
        unsquashfs -d /snap/go/current go.snap && \
        cd /snap/bin && \
        ln -s /snap/go/current/bin/go

FROM ubuntu:xenial
COPY --from=builder /snap /snap

RUN apt clean && \
	apt update && \
        apt dist-upgrade --yes && \
        apt install --yes \
        sudo locales \
        gcc git && \
        locale-gen en_US.UTF-8

# Set the proper environment
ENV LANG en_US.UTF-8
ENV LANGUAGE en_US:en
ENV LC_ALL en_US.UTF-8
ENV PATH "/snap/bin:$PATH"
ENV SNAP /snap/snapcraft/current
ENV SNAP_NAME snapcraft
ENV SNAP_ARCH amd64

SHELL ["/bin/bash", "-c"]
ENTRYPOINT "snapcraft"



